prov/verbs/src/verbs_msg_ep.c

```diff
@@ -221,7 +225,18 @@ static int fi_ibv_msg_ep_enable(struct fid_ep *ep)
 
 	attr.cap.max_inline_data = _ep->info->tx_attr->inject_size;
 
-	attr.srq = NULL;
+	if (_ep->srq_ep) {
+		attr.srq =_ep->srq_ep->srq;
+		/* Use of SRQ, no need to allocate recv_wr entries in the QP */
+		attr.cap.max_recv_wr = 0;
+
+		/* Cannot post a recv WR to a QP where a SRQ is attached to */
+		verbs_ops_msg = fi_ibv_msg_ep_ops_msg(_ep);
+		verbs_ops_msg->recv = fi_no_msg_recv;
+		verbs_ops_msg->recvv = fi_no_msg_recvv;
+		verbs_ops_msg->recvmsg = fi_no_msg_recvmsg;
```

Comment:
Sorry for the delay, I missed your comment concerning this block of code.
I decided to create a new 'struct fi_ops_msg' where all the recv* operations redirect to ENOSYS versions. With this solution, there is no need to check whether the endpoint uses SRX  or not.
The new code has been pushed into Github.